contracts/src/utils/Counter.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.20;
+
+library Counter {
+    function createCounter(uint256 length) internal pure returns (uint256[] memory counter) {
+        uint256 counterLength = length / 16 + (length % 16 == 0 ? 0 : 1);
+        counter = new uint256[](counterLength);
+    }
+
+    function get(uint256[] storage self, uint256 index) internal view returns (uint16) {
+        uint256 element = index >> 4;
+        uint8 inside = uint8(index) & 0x0F;
+        return uint16((self[element] >> (16 * inside)) & 0xFFFF);
+    }
+
+    function set(uint256[] storage self, uint256 index, uint16 value) internal {
+        uint256 element = index >> 4;
+        uint8 inside = uint8(index) & 0x0F;
+        uint256 zero = ~(uint256(0xFFFF) << (16 * inside));
```

Comment:
Do you have an intuitive sense what these operations and magic constants mean? If not, lets try make sense of it, and then add documentation/comments.

Reply:
Addressed in https://github.com/Snowfork/snowbridge/pull/956/commits/af16c9598bd6914ec1e93f7454e204a69ad49594